app/src/main/java/org/mozilla/focus/session/SessionManager.kt

```diff
@@ -0,0 +1,227 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.session
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.text.TextUtils
+import org.mozilla.focus.architecture.NonNullLiveData
+import org.mozilla.focus.architecture.NonNullMutableLiveData
+import org.mozilla.focus.customtabs.CustomTabConfig
+import org.mozilla.focus.shortcut.HomeScreen
+import org.mozilla.focus.utils.SafeIntent
+import org.mozilla.focus.utils.UrlUtils
+import java.util.*
+
+/**
+ * Sessions are managed by this global SessionManager instance.
+ */
+class SessionManager private constructor() {
+
+    private val sessions: NonNullMutableLiveData<List<Session>>
+    private var currentSessionUUID: String? = null
+
+    /**
+     * Get the current session. This method will throw an exception if there's no active session.
+     */
+    val currentSession: Session
+        get() {
+            if (currentSessionUUID == null) {
+                throw IllegalAccessError("There's no active session")
+            }
+
+            return getSessionByUUID(currentSessionUUID!!)
+        }
+
+    val numberOfSessions: Int
+        get() = sessions.value.size
+
+    val positionOfCurrentSession: Int
+        get() {
+            return getPositionByUUID(currentSessionUUID)
+        }
+
+    init {
+        this.sessions = NonNullMutableLiveData(
+                Collections.unmodifiableList(emptyList()))
+    }
+
+    /**
+     * Handle this incoming intent (via onCreate()) and create a new session if required.
+     */
+    fun handleIntent(context: Context, intent: SafeIntent, savedInstanceState: Bundle?) {
+        if (intent.flags and Intent.FLAG_ACTIVITY_LAUNCHED_FROM_HISTORY != 0) {
+            // This Intent was launched from history (recent apps). Android will redeliver the
+            // original Intent (which might be a VIEW intent). However if there's no active browsing
+            // session then we do not want to re-process the Intent and potentially re-open a website
+            // from a session that the user already "erased".
+            return
+        }
+
+        if (savedInstanceState != null) {
+            // We are restoring a previous session - No need to handle this Intent.
+            return
+        }
+
+        createSessionFromIntent(context, intent)
+    }
+
+    /**
+     * Handle this incoming intent (via onNewIntent()) and create a new session if required.
+     */
+    fun handleNewIntent(context: Context, intent: SafeIntent) {
+        createSessionFromIntent(context, intent)
+    }
+
+    private fun createSessionFromIntent(context: Context, intent: SafeIntent) {
+        val action = intent.action
+
+        if (Intent.ACTION_VIEW == action) {
+            val dataString = intent.dataString
+            if (TextUtils.isEmpty(dataString)) {
```

Comment:
I think Kotlin has a extension function for that. Something like isNullOrEmpty()